src/Conversion/ONNXToTorch/NN/ReduceMean.cpp

```diff
@@ -60,27 +60,30 @@ struct ONNXReduceMeanOpToTorchLowering : public ConversionPattern {
     ONNXReduceMeanOp reduceMean = llvm::dyn_cast_or_null<ONNXReduceMeanOp>(op);
     assert(reduceMean && "Expecting op to have a strong type");
 
-    // **TODO**: reduceMean.axes is not being used currently. Probably use
-    // AtenMeanDimOp instead of AtenMean, or use both.
-
     mlir::MLIRContext *context = reduceMean.getContext();
     Location loc = reduceMean.getLoc();
 
     auto axis = mlir::extractFromI64ArrayAttr(reduceMean.axesAttr());
     if(!(axis.size() == 2 && axis[0] == 2 && axis[1] == 3))
-      op->emitError("Not implemented yet for general axis sizes");
+      assert("Not implemented yet for general axis sizes");
```

Comment:
Does `assert` result in a more graceful failure than `op->emitError`? I had expected the opposite.

Of note also is that an assert is for sanity checking, as in you do not expect it to be possible/legal to create the condition that is being asserted, which is why optimized builds can delete asserts. If legal onnx source could have different values here, and we want to fail to convert .

Reply:
Just had a read through other conversion passes. Standard practice seems to be `op->emitError` and failure in this case.